run_meta_analysis.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# coding: utf-8
+import argparse
+import hail as hl
+from ukbb_pan_ancestry import *
+from ukb_common import *
+
+
+def all_and_leave_one_out(x, pop, all_f=hl.sum, loo_f=lambda i, x: hl.sum(x) - hl.or_else(x[i], 0)):
+    arr = hl.array([all_f(x)])
+    arr = arr.extend(hl.map(lambda i: loo_f(i, x), hl.range(hl.len(pop))))
+    return arr
+
+
+def main(args):
+    hl.init(log='/run_meta_analysis.log')
+
+    # Read in all sumstats
+    mt = hl.read_matrix_table('gs://ukb-diverse-pops/combined_results/all_sumstats.mt')
+
+    # Run meta-analysis (all + leave-one-out)
+    mt = mt.annotate_entries(unnorm_beta=mt.summary_stats.BETA / (mt.summary_stats.SE**2),
+                             inv_se2=1 / (mt.summary_stats.SE**2))
+
+    mt = mt.transmute_entries(sum_unnorm_beta=all_and_leave_one_out(mt.unnorm_beta, mt.pheno_data.pop),
+                              sum_inv_se2=all_and_leave_one_out(mt.inv_se2, mt.pheno_data.pop))
+
+    mt = mt.transmute_entries(META_BETA=mt.sum_unnorm_beta / mt.sum_inv_se2,
+                              META_SE=hl.map(lambda x: hl.sqrt(1 / x), mt.sum_inv_se2))
+
+    mt = mt.annotate_entries(META_BETA=hl.map(lambda x: hl.or_missing(hl.is_finite(x), x), mt.META_BETA),
+                             META_SE=hl.map(lambda x: hl.or_missing(hl.is_finite(x), x), mt.META_SE))
+
+    mt = mt.annotate_entries(META_Pvalue=hl.map(lambda x: 2 * hl.pnorm(x), -hl.abs(mt.META_BETA / mt.META_SE)))
+
+    # Add other annotations
+    mt = mt.annotate_entries(variant_exists=hl.map(lambda x: ~hl.is_missing(x), mt.summary_stats.BETA),
+                             af_cases=hl.map(lambda x: x["AF.Cases"] * x.N, mt.summary_stats),
```

Comment:
Is this AC or AF? Maybe I missed it, but I don't see where you divide back by N to get `META_AF_Cases`?

Reply:
Good catch! Fixed now